packages/runtime/container-runtime/src/summarizerClientElection.ts

```diff
@@ -0,0 +1,83 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IEventProvider, ITelemetryLogger } from "@fluidframework/common-definitions";
+import { MessageType } from "@fluidframework/protocol-definitions";
+import { OrderedClientElection, ITrackedClient } from "./orderedClientElection";
+import { ISummaryCollectionOpEvents } from "./summaryCollection";
+
+export const summarizerClientType = "summarizer";
+
+/**
+ * This class encapsulates logic around tracking the elected summarizer client.
+ * It will handle updated the elected client when a summary ack hasn't been seen
+ * for some configured number of ops.
+ */
+export class SummarizerClientElection {
+    /** Used to calculate number of ops since last summary ack for the current elected client */
+    private lastSummaryAckSeqForClient = 0;
+    private _hasSummarizersInQuorum: boolean;
+    private hasLoggedTelemetry = false;
+
+    public get electedClientId() {
+        return this.clientElection.getElectedClient()?.clientId;
+    }
+
+    public get hasSummarizersInQuorum() {
+        return this._hasSummarizersInQuorum;
+    }
+
+    constructor(
+        private readonly logger: ITelemetryLogger,
+        private readonly summaryCollection: IEventProvider<ISummaryCollectionOpEvents>,
+        public readonly clientElection: OrderedClientElection,
+        private readonly maxOpsSinceLastSummary: number,
+        private readonly refreshSummarizer: () => void,
+    ) {
+        this.summaryCollection.on("default", (op) => {
+            const opsSinceLastAckForClient = op.sequenceNumber - this.lastSummaryAckSeqForClient;
+            if (
+                opsSinceLastAckForClient > this.maxOpsSinceLastSummary
+                && !this.hasLoggedTelemetry
+                && this.electedClientId !== undefined
+            ) {
+                // Limit telemetry to only next client?
+                this.logger.sendErrorEvent({
+                    eventName: "ElectedClientNotSummarizing",
+                    electedClientId: this.electedClientId,
+                    sequenceNumber: op.sequenceNumber,
```

Comment:
>: [](http://example.com/codeflow?start=34&length=1)

no need to log seq. we log that by default  up the chain of loggers: [https://github.com/microsoft/FluidFramework/blob/6e977e8e738fb68717317991615fc35375d6f04a/packages/loader/container-loader/src/container.ts#L604](https://github.com/microsoft/FluidFramework/blob/6e977e8e738fb68717317991615fc35375d6f04a/packages/loader/container-loader/src/container.ts#L604)